sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs

```diff
@@ -0,0 +1,520 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Azure.Messaging.ServiceBus.Primitives;
+
+namespace Azure.Messaging.ServiceBus
+{
+    /// <summary>
+    /// The message object used to communicate and transfer data with Service Bus.
+    /// </summary>
+    /// <remarks>
+    /// The message structure is discussed in detail in the <a href="https://docs.microsoft.com/azure/service-bus-messaging/service-bus-messages-payloads">product documentation.</a>
+    /// </remarks>
+    public class ServiceBusMessage
```

Comment:
`BrokeredMessage` -> `Message` -> `ServiceBusMessage`. History rhymes.

Reply:
On a serious note, what's the thought against keeping it a `Message`?

Reply:
My thought was that `Message` is too generic. We are also considering having two model types for sending and receiving a message so if we do that we would need to come up with new names.

Reply:
That was my argument in the previous round. The idea of different types of sent and received messages is interesting, but I'd need to see good reasons to decide for or against it. I can see where a sent message has no broker-set properties and therefore could be slimmer version. On the other hand, it is still a brokered message, wherever it was received or constructed in the user code.

Reply:
Yes, the idea is that we can avoid cluttering up the send model message with broker-set properties. 
This is captured in https://github.com/Azure/azure-sdk-for-net/issues/9886